### PR TITLE
feat: make it possible to disable NewRelic globally

### PR DESCRIPTION
**Pull request by regisb**
By defining the `ENABLE_NEW_RELIC='false'` environment variable, the new relic
logging integration will be disabled, which is a must-have for most non-edX
platforms.

Close https://github.com/openedx/frontend-wg/issues/14

cc @davidjoy.

### CODE REVIEW

**Comment by edx-webhook** [comment]
Thanks for the pull request, @regisb! I've created [OSPR-6240](https://openedx.atlassian.net/browse/OSPR-6240) to keep track of it in JIRA, where we prioritize reviews. Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.

File: config/webpack.prod.config.js [hunk 1]
```diff
@@ -172,26 +194,10 @@ module.exports = merge(commonConfig, {
...
-      applicationID: process.env.NEW_RELIC_APP_ID || 'undefined_application_id',
-    }),
-    new NewRelicSourceMapPlugin({
-      applicationId: process.env.NEW_RELIC_APP_ID,
-      apiKey: process.env.NEW_RELIC_ADMIN_KEY,
-      staticAssetUrl: process.env.BASE_URL,
-      // upload source maps in prod builds only
-      noop: typeof process.env.NEW_RELIC_ADMIN_KEY === 'undefined',
-    }),
+  ].concat(extraPlugins).concat([
```
Comment (davidjoy) [inline config/webpack.prod.config.js:197]: Readability nit: Might suggest replacing this line with just:

```
...extraPlugins,
```

And leaving all the plugins in the same array.  The spread operator will just splat the extra ones into the array.  And I think we can also move `BundleAnalyzerPlugin` above the extra ones, which will help it feel a bit more organized.

Comment (regisb) [reply config/webpack.prod.config.js]: TIL the spread operator!

**Comment by natabene** [comment]
@regisb Thank you for your contribution. We will review this as soon as we can.

**Review by macdiesel — APPROVED** [review]

**Review by manny-m — APPROVED** [review]
👍

**Comment by regisb** [comment]
Thanks for the reviews! Can we merge this and make a release? We will then have to upgrade frontend-build in the account, ecommerce gradebook, learning, payment, profile MFEs.

**Comment by edx-webhook** [comment]
@regisb 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=edx/frontend-build&entry.867055334=https://github.com/edx/frontend-build/pull/215&entry.1484655318=https://github.com/regisb&entry.752974735=2021-11-25+22:54&entry.1917517419=2021-12-08+16:50&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**